### PR TITLE
Fixed incorrect format argument

### PR DESCRIPTION
**Pull request by jeanbovet**
@wmathurin @bhariharan

### CODE REVIEW

**Comment by bhariharan** [comment]
3 failures (seem unrelated):
```
[14:38:34]: ▸ SalesforceRestAPITests
[14:38:34]: ▸ testUpsert, ((listener.returnStatus) equal to (kTestRequestStatusDidLoad)) failed: ("didFail") is not equal to ("didLoad") - request should have succeeded
[14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m:660
[14:38:34]: ▸ ```
[14:38:34]: ▸      listener = [self sendSyncRequest:secondUpsertRequest];
[14:38:34]: ▸      XCTAssertEqualObjects(listener.returnStatus, kTestRequestStatusDidLoad, @"request should have succeeded");
[14:38:34]: ▸ 
[14:38:34]: ▸   ```
[14:38:34]: ▸ testUpsert, ((secondRetrievedName) equal to (accountNameUpdated)) failed: ("RestClientTestsiOS580426701.041461") is not equal to ("RestClientTestsiOS580426701.041461_updated") - wrong name retrieved
[14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m:667
[14:38:34]: ▸ ```
[14:38:34]: ▸      NSString *secondRetrievedName = ((NSDictionary *) listener.dataResponse)[NAME];
[14:38:34]: ▸      XCTAssertEqualObjects(secondRetrievedName, accountNameUpdated, "wrong name retrieved");
[14:38:34]: ▸  }
[14:38:34]: ▸   ```
[14:38:34]: ▸ SalesforceSDKIdentityTests
[14:38:34]: ▸ testRetrieveIdentitySuccess, ((idData.firstName) != nil) failed - firstName should not be nil.
[14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKIdentityTests.m:118
[14:38:34]: ▸ ```
[14:38:34]: ▸     XCTAssertNotNil(idData.email, @"email should not be nil.");
[14:38:34]: ▸     XCTAssertNotNil(idData.firstName, @"firstName should not be nil.");
[14:38:34]: ▸     XCTAssertNotNil(idData.lastName, @"lastName should not be nil.");
[14:38:34]: ▸   ```
[14:38:34]: ▸ 	 Executed 316 tests, with 3 failures (0 unexpected) in 196.486 (197.564) seconds
[14:38:34]: ▸ 
[14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 224.949 elapsed -- Testing started completed.
[14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
[14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 224.949 sec, +224.949 sec -- end
[14:38:39]: ▸ Failing tests:
[14:38:39]: ▸ -[SalesforceRestAPITests testUpsert]
[14:38:39]: ▸ -[SalesforceRestAPITests testUpsert]
[14:38:39]: ▸ -[SalesforceSDKIdentityTests testRetrieveIdentitySuccess]
[14:38:39]: ▸ ** TEST FAILED **
```

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m [hunk 1]
```diff
@@ -1438,7 +1438,7 @@ - (void)retrieveUserPhotoIfNeeded:(SFUserAccount *)account {
         SFNetwork *network = [[SFNetwork alloc] initWithEphemeralSession];
         [network sendRequest:request  dataResponseBlock:^(NSData *data, NSURLResponse *response, NSError *error){
             if (error) {
-                [SFSDKCoreLogger w:[self class] format:@"Error while trying to retrieve user photo: %@ %@", (long) error.code, error.localizedDescription];
+                [SFSDKCoreLogger w:[self class] format:@"Error while trying to retrieve user photo: %ld %@", (long) error.code, error.localizedDescription];
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: Or the other way it to do:
```objective-c
[SFSDKCoreLogger w:[self class] format:@"Error while trying to retrieve user photo: %@ %@", @(error.code), error.localizedDescription];
```

Comment (jeanbovet) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m]: True! I did that to align with similar logging in this file but happy to change it next time I'm visiting that file :-)

**Comment by opas2536** [comment]
> 3 failures (seem unrelated):
> 
> ```
> [14:38:34]: ▸ SalesforceRestAPITests
> [14:38:34]: ▸ testUpsert, ((listener.returnStatus) equal to (kTestRequestStatusDidLoad)) failed: ("didFail") is not equal to ("didLoad") - request should have succeeded
> [14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m:660
> [14:38:34]: ▸ ```
> [14:38:34]: ▸      listener = [self sendSyncRequest:secondUpsertRequest];
> [14:38:34]: ▸      XCTAssertEqualObjects(listener.returnStatus, kTestRequestStatusDidLoad, @"request should have succeeded");
> [14:38:34]: ▸ 
> [14:38:34]: ▸   ```
> [14:38:34]: ▸ testUpsert, ((secondRetrievedName) equal to (accountNameUpdated)) failed: ("RestClientTestsiOS580426701.041461") is not equal to ("RestClientTestsiOS580426701.041461_updated") - wrong name retrieved
> [14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m:667
> [14:38:34]: ▸ ```
> [14:38:34]: ▸      NSString *secondRetrievedName = ((NSDictionary *) listener.dataResponse)[NAME];
> [14:38:34]: ▸      XCTAssertEqualObjects(secondRetrievedName, accountNameUpdated, "wrong name retrieved");
> [14:38:34]: ▸  }
> [14:38:34]: ▸   ```
> [14:38:34]: ▸ SalesforceSDKIdentityTests
> [14:38:34]: ▸ testRetrieveIdentitySuccess, ((idData.firstName) != nil) failed - firstName should not be nil.
> [14:38:34]: ▸ /Users/distiller/SalesforceMobileSDK-iOS/libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKIdentityTests.m:118
> [14:38:34]: ▸ ```
> [14:38:34]: ▸     XCTAssertNotNil(idData.email, @"email should not be nil.");
> [14:38:34]: ▸     XCTAssertNotNil(idData.firstName, @"firstName should not be nil.");
> [14:38:34]: ▸     XCTAssertNotNil(idData.lastName, @"lastName should not be nil.");
> [14:38:34]: ▸   ```
> [14:38:34]: ▸ 	 Executed 316 tests, with 3 failures (0 unexpected) in 196.486 (197.564) seconds
> [14:38:34]: ▸ 
> [14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 224.949 elapsed -- Testing started completed.
> [14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
> [14:38:39]: ▸ 2019-05-24 14:38:39.251 xcodebuild[3127:9325] [MT] IDETestOperationsObserverDebug: 224.949 sec, +224.949 sec -- end
> [14:38:39]: ▸ Failing tests:
> [14:38:39]: ▸ -[SalesforceRestAPITests testUpsert]
> [14:38:39]: ▸ -[SalesforceRestAPITests testUpsert]
> [14:38:39]: ▸ -[SalesforceSDKIdentityTests testRetrieveIdentitySuccess]
> [14:38:39]: ▸ ** TEST FAILED **
> ```

**Comment by opas2536** [comment]
Duplicate of #

**Pull request merged**